src/components/Editor/Editor.ts

```diff
@@ -0,0 +1,102 @@
+import debounce from "lodash.debounce";
+
+import { createComponent, useEffect, useState } from "@/core";
+import { ChildDocumentLinks, NotFound } from "@/components";
+import { getDocument } from "@/apis";
+import { DocumentMetaDto, DocumentPutRequestDto, DocumentPutResponseDto } from "@/types";
+import styles from "./editor.module.scss";
+
+const { s_editorForm, s_editorInput, s_editorContent, s_childDocuments } = styles;
+
+interface EditorProps {
+  documentId: number;
+  modifyDocument: ({ id, title, content }: DocumentPutRequestDto) => Promise<DocumentPutResponseDto>;
+}
+
+function Editor({ documentId, modifyDocument }: EditorProps) {
+  const [childDocuments, setChildDocuments] = useState<DocumentMetaDto[]>([]);
+  const [documentForm, setDocumentForm] = useState({ title: "", content: "" });
+  const [isDocumentNotFound, setIsDocumentNotFound] = useState(false);
+
+  useEffect(() => {
+    const fetchDocument = async (id: number) => {
+      try {
+        const { title, content, documents } = await getDocument(id);
+
+        setDocumentForm({ title, content: content ?? "" });
+        setChildDocuments(documents);
+      } catch (error) {
+        console.error(error);
+        setIsDocumentNotFound(true);
+      }
+    };
+
+    fetchDocument(documentId);
+  }, [documentId]);
+
+  const debouncedUpdate = debounce(async (title, content) => {
+    try {
+      const modifiedDocument = await modifyDocument({ id: documentId, title, content });
+
+      setDocumentForm({ title: modifiedDocument.title, content: modifiedDocument.content });
```

Comment:
title, content가 동일한 이름으로 두 군데에서 사용되어서 modifiedDocument 부분은 구조분해 할당을 하지 못한 부분이 아쉽네요 😢

Reply:
naming 관련해서 좋은 이름이 있다면 추천해주세요! : )

Reply:
Promise chaining의 then에서 구조 분해 할당을 하면 modifiedDocument를 선언하지 않아도 될 것 같다는 생각이 들어요. 좋은 이름으로 바꾸는 방법은 아직 잘 모르겠습니다 😂

Reply:
바로 title, content로 구조 분해 할당을 하지 못하는 이유는 아래 코드에서 볼 수 있듯이 `debouncedUpdate`에서 파라미터의 이름을 title, content로 받기 때문에 이름이 겹쳐 사용하지 못하기 때문입니다. `then`에서 처리를 해주더라도 사용할 수 없어요. 그래서 파라미터 이름으로 좋은 이름이 있는지 여쭤본 것입니다 : )
```ts
  const debouncedUpdate = debounce(async (title, content) => {
    try {
      const modifiedDocument = await modifyDocument({ id: documentId, title, content });

      setDocumentForm({ title: modifiedDocument.title, content: modifiedDocument.content });
    } catch (error) {
      console.error(error);
    }
  }, 1000);
```

Reply:
```suggestion
      modifyDocument({ id: documentId, title, content })
      .then(({ title, content }) => {
        setDocumentForm({ title, content });
      })
```
스코프가 달라지면서 구조 분해 할당이 가능한 것 같습니다. await의 경우는 가독성이 좋다는 장점이 있어서 선호하시는 스타일에 따라 사용하시면 될 것 같아요 👍

Reply:
아 여기서 then을 사용하는 거였군요! 흠 저는 디스트럭처링 하여 쓰기 위해 then을 사용하는 방법이 좀 꺼려지네요. 코드가 깊어질수록 가독성이 떨어지기 때문입니다. then을 사용해서 코드 깊이가 깊어지는것보다 파라미터 이름을 변경하는 쪽이 좋을 것 같다는 생각이 드는데, 어떻게 생각하시나요?